tests/testthat/test-recode_into.R

```diff
@@ -0,0 +1,81 @@
+test_that("recode_into", {
+  x <<- 1:10
+  out <- recode_into(
+    x > 5 ~ "a",
+    x > 2 & x <= 5 ~ "b",
+    default = "c"
+  )
+  expect_identical(out, c("c", "c", "b", "b", "b", "a", "a", "a", "a", "a"))
+})
+
+test_that("recode_into, check mixed types", {
+  x <<- 1:10
+  expect_error(out <- recode_into( # nolint
+    x > 5 ~ 1,
+    x > 2 & x <= 5 ~ "b"
+  ))
```

Comment:
Last thing for me, can you add the regexp for `expect_error` / `_warning` / `_message`?

After that and once the examples are fixed I'm ok to merge